src/config/config.ts

```diff
@@ -805,10 +805,24 @@ export class Config implements IConfig {
     return commandPlugins[0]
   }
 
+  /**
+    * Insert legacy plugins
+    *
+    * Replace invalid CLI plugins (cli-engine plugins, mostly Heroku) loaded via `this.loadPlugins`
+    * with oclif-compatible ones returned by @oclif/plugin-legacy init hook.
+    *
+    * @param plugins array of oclif-compatible plugins
+    * @returns void
+    */
   private insertLegacyPlugins(plugins: IPlugin[]) {
     for (const plugin of plugins) {
       const idx = this.plugins.findIndex(p => p.name === plugin.name)
-      if (idx !== -1) this.plugins = this.plugins.splice(idx, 1, plugin)
```

Comment:
`Array.splice` mutates the array and returns an array of deleted items in this case, see:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice#return_value

so, `this.plugins` was set to the non-compatible version of the plugin loaded here:
https://github.com/oclif/core/blob/a2efc040855969678a6aaf14d3ceb1d25d7a8411/src/config/config.ts#L193